view/src/storage.rs

```diff
@@ -666,6 +697,97 @@ impl Storage {
         Ok(())
     }
 
+    pub async fn give_advice(&self, note: Note) -> anyhow::Result<()> {
+        let mut tx = self.pool.begin().await?;
+
+        let note_commitment = note.commit().0.to_bytes().to_vec();
+        let address = note.address().to_vec();
+        let amount = u64::from(note.amount()) as i64;
+        let asset_id = note.asset_id().to_bytes().to_vec();
+        let blinding_factor = note.note_blinding().to_bytes().to_vec();
+
+        sqlx::query!(
+            "INSERT INTO notes
+                    (
+                        note_commitment,
+                        address,
+                        amount,
+                        asset_id,
+                        blinding_factor
+                    )
+                    VALUES
+                    (?, ?, ?, ?, ?)",
+            note_commitment,
+            address,
+            amount,
+            asset_id,
+            blinding_factor,
+        )
+        .execute(&mut tx)
+        .await?;
+
+        tx.commit().await?;
+
+        Ok(())
+    }
+
+    /// Return advice about note contents for use in scanning.
+    ///
+    /// Given a list of note commitments, this method checks whether any of them
+    /// correspond to notes that have been recorded in the database but not yet
+    /// observed during scanning.
+    pub async fn scan_advice(
+        &self,
+        note_commitments: Vec<note::Commitment>,
+    ) -> anyhow::Result<BTreeMap<note::Commitment, Note>> {
```

Comment:
This method is how the view server gets advice to use while scanning a compact block: it builds a list of all the rolled-up note commitments, then has the `Storage` cross-reference with known notes.